task-2/src/Inventory.ts

```diff
@@ -0,0 +1,21 @@
+import { Item } from "./Item";
+import { ItemComparator } from "./ItemComparator";
+
+export class Inventory {
+    private items: Item[];
+
+    addItem(item: Item): void {
+        this.items.push(item);
+    }
+
+    sort(comparator?: ItemComparator): void {
+        const defaultComparator = function (a: Item, b: Item): number {
+            return a.compareTo(b);
+        }
+        this.items.sort(comparator?.compare || defaultComparator);
```

Comment:
Minor: when you check on null, and when your ECMAScript version allows it, it's better to use `??` over `||`